modules/light-clients/07-celestia/client_state.go

```diff
@@ -0,0 +1,125 @@
+package celestia
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	ibcerrors "github.com/cosmos/ibc-go/v8/modules/core/errors"
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+	ibctm "github.com/cosmos/ibc-go/v8/modules/light-clients/07-tendermint"
+)
+
+var _ exported.ClientState = (*ClientState)(nil)
+
+// ClientType implements exported.ClientState.
+func (*ClientState) ClientType() string {
+	return ModuleName
+}
+
+// GetLatestHeight implements exported.ClientState.
+func (cs *ClientState) GetLatestHeight() exported.Height {
+	return cs.BaseClient.GetLatestHeight()
+}
+
+// GetTimestampAtHeight implements exported.ClientState.
+func (cs *ClientState) GetTimestampAtHeight(ctx sdk.Context, clientStore storetypes.KVStore, cdc codec.BinaryCodec, height exported.Height) (uint64, error) {
+	return cs.BaseClient.GetTimestampAtHeight(ctx, clientStore, cdc, height)
+}
+
+// Status implements exported.ClientState.
+func (cs *ClientState) Status(ctx sdk.Context, clientStore storetypes.KVStore, cdc codec.BinaryCodec) exported.Status {
+	return cs.BaseClient.Status(ctx, clientStore, cdc)
+}
+
+// Initialize implements exported.ClientState.
+func (cs *ClientState) Initialize(ctx sdk.Context, cdc codec.BinaryCodec, clientStore storetypes.KVStore, consensusState exported.ConsensusState) error {
+	return cs.BaseClient.Initialize(ctx, cdc, clientStore, consensusState)
+}
+
+// Validate implements exported.ClientState.
+func (cs *ClientState) Validate() error {
+	return cs.BaseClient.Validate()
+}
+
+// VerifyMembership implements exported.ClientState.
+// TODO: Revise and look into delay periods for this.
```

Comment:
I think we can add a `keeper` pkg and control delay period values with gov gated on-chain params.

Reply:
https://github.com/cosmos/ibc-go/issues/5852